src/supplemental/nanolib/test_conf/nmq_test.conf

```diff
@@ -1063,3 +1138,64 @@ rules.mysql.mysql_rule_db {
 		}
 	]
 }
+
+# # #====================================================================
+# # # Exchange configuration for Embedded Messaging Queue
+# # #====================================================================
+# # # Initalize multiple MQ exchanger by giving them different name (mq1)
+# exchange_client.mq1 {
+# 	# # exchanges contains multiple MQ exchanger
+# 	exchange {
+# 		# # MQTT Topic for filtering messages and saving to queue
+# 		topic = "exchange/topic1",
+# 		# # MQ name
+# 		name = "exchange_no1",
+# 		# # MQ category. Only support Ringbus for now
+# 		ringbus = {
+# 			# # ring buffer name
+# 			name = "ringbus",
+# 			# # max length of ring buffer (msg count)
+# 			cap = 1000
+# 		}
+# 	}
+# }
+
+
+# # #====================================================================
+# # # Parquet configuration (Apply to Exchange/Messaging_Queue)
+# # #====================================================================
+# parquet {
+# 	# # Parquet compress type.
+# 	# #
+# 	# # Value: uncompressed | snappy | gzip | brotli | zstd | lz4
+# 	compress = uncompressed
+# 	# # Encryption options
+# 	encryption {
+# 		# # Set a key retrieval metadata.
+# 		# #
+# 		# # Value: String
+# 		key_id = kf
+# 		# # Parquet encryption key.
+# 		# #
+# 		# # Value: String key must be either 16, 24 or 32 bytes.
+# 		key = "0123456789012345"
+# 		# # Set encryption algorithm. If not called, files 
+# 		# # will be encrypted with AES_GCM_V1 (default).
+# 		# #
+# 		# # Value: AES_GCM_CTR_V1 | AES_GCM_V1
+# 		type = AES_GCM_V1
+# 	}
+# 	# # The dir for parquet files.
+# 	# #
+# 	# # Value: Folder
+# 	dir = "/tmp/nanomq-parquet"
+# 	# # The prefix of parquet files written.
+# 	# #
+# 	# # Value: string
+# 	file_name_prefix = ""
+# 	# # Maximum rotation count of parquet files.
+# 	# #
+# 	# # Value: Number
+# 	# # Default: 5
+# 	file_count = 5
+# }
```

Comment:
There are plenty of memleaks in exchange & parquet now.

Reply:
worth an issue

Reply:
> There are plenty of memleaks in conf parse in exchange & parquet now.

I have made some updates. Those memory leakage will no longer exists after that pr was merged.